cookbooks/bcpc-hadoop/recipes/copylog.rb

```diff
@@ -0,0 +1,70 @@
+#
+# Recipe to copy log data from nodes into HDFS
+# Flume is used to copy data and runs as an agent on the nodes
+# 
+#**** Attention ***** Attention ****
+# This should be the last recipe in the runlist for any role since 
+# individual recipes can make requests to copy log files to HDFS
+# This can be achieved by adding the Copylog role as the last role to the node
+# Since flume writes into HDFS nodes running this recipe should have HDFS
+# client components installed on them.
+
+%w{flume flume-agent}.each do |p|
+  package p do
+    action :upgrade
+  end
+end
+
+bash "make_shared_logs_dir" do
+  code <<EOH
+  hdfs dfs -mkdir -p #{node['bcpc']['hadoop']['hdfs_url']}/user/flume/logs/ && \
+  hdfs dfs -chown -R flume #{node['bcpc']['hadoop']['hdfs_url']}/user/flume/
+EOH
+  user "hdfs"
+  not_if "hdfs dfs -test #{node['bcpc']['hadoop']['hdfs_url']}/user/flume/logs/", :user => "hdfs"
+end
+
+service "flume-agent" do
+  supports :status => true, :restart => true, :reload => false
+  action [:stop]
```

Comment:
Do we want to disable too? It seems this might be a case for disabling this package via dpkg_autostart?